server/mailet/mailets/src/main/java/org/apache/james/transport/matchers/IsMarkedAsSpam.java

```diff
@@ -51,9 +54,18 @@ public class IsMarkedAsSpam extends GenericMatcher {
 
     private static final String YES = "yes";
 
+    private String spamStatusHeader = SpamAssassinResult.STATUS_MAIL.asString();
+
     @Override
     public String getMatcherInfo() {
-        return "Has org.apache.james.spamassassin.status per recipient header with a Yes value";
+        return "Has " + spamStatusHeader + " per recipient header with a Yes value";
+    }
+
+    @Override
+    public void init() throws MessagingException {
+        spamStatusHeader = Optional.ofNullable(getCondition())
+            .filter(s -> !s.isEmpty())
+            .orElse(SpamAssassinResult.STATUS_MAIL.asString());
     }
```

Comment:
Document this change (javadoc above, in `server/apps/distributed-app/docs/modules/ROOT/partials/IsMarkedAsSpam.adoc`)

Reply:
Your mean is do not use `SpamAssassinResult.STATUS_MAIL` by default?

IMO, we should keep it, it won't make degrade the old configure when not provided status package